core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1259,12 +1260,18 @@ class ReplicaManager(val config: KafkaConfig,
                     .setIsFutureKey(log.isFuture)
                 }.toList.asJava)
             }.filterNot(_.partitions().isEmpty).toList.asJava
-
-            createDescribeLogDirsResult(absolutePath, totalBytes, usableBytes, topicInfos)
           case None =>
-            createDescribeLogDirsResult(absolutePath, totalBytes, usableBytes, Collections.emptyList())
+            Collections.emptyList[DescribeLogDirsTopic]()
         }
 
+        val describeLogDirsResult = new DescribeLogDirsResponseData.DescribeLogDirsResult()
+          .setLogDir(absolutePath)
+          .setErrorCode(Errors.NONE.code)
+          .setTotalBytes(totalBytes).setUsableBytes(usableBytes)
+        if (!topicInfos.isEmpty)
```

Comment:
Is this check necessary?

Reply:
I think you are right, In `DescribeLogDirsResult` contructor has already init a empty List, so I didn't need to check `!topicInfos.isEmpty`